torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -2052,6 +2052,34 @@ void verify_params_across_processes(
     const c10::intrusive_ptr<c10d::ProcessGroup>& process_group,
     const std::vector<at::Tensor>& params,
     const c10::optional<std::weak_ptr<c10d::Logger>>& logger) {
+
+  // First verify number of parameters to avoid inconsistent inputs into
+  // broadcast which can cause a crash.
+  // See https://github.com/pytorch/pytorch/issues/73547
+
+  at::TensorOptions param_size_options;
+  param_size_options = param_size_options.dtype(at::kLong);
+  param_size_options = param_size_options.device(params[0].device());
+  // Note: Not using tensor building API because of
+  // https://github.com/pytorch/pytorch/issues/74114
+  at::Tensor param_size_tensor = at::tensor(
+    {static_cast<int64_t>(params.size())}, param_size_options);
+
+  // Broadcast and verify parameter size.
+  std::vector<at::Tensor> param_size_vec{param_size_tensor};
+  process_group->broadcast(param_size_vec)->wait();
+  auto res = param_size_tensor[0].item<int>();
+  TORCH_CHECK(
```

Comment:
curious, does this mean some DDP process will crash, while other won't if their size matches? If that's the case, does it mean allgather might be better, as all processes can make the same decision here and throw the same error?

Reply:
This is a good point, it would be a lot better if all processes make the same decision. Next update will fix this